src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -2054,4 +2054,16 @@ Widening and user defined conversions are not supported with generic types.</val
   <data name="PreferIsEmptyOverAnyMessage" xml:space="preserve">
     <value>Prefer an 'IsEmpty' check rather than using 'Any()', both for clarity and for performance</value>
   </data>
+  <data name="PreferDictionaryTryAddValueCodeFixTitle" xml:space="preserve">
+    <value>Use 'TryAdd(TKey, TValue)'</value>
+  </data>
+  <data name="PreferDictionaryTryAddTitle" xml:space="preserve">
+    <value>Prefer the 'IDictionary.TryAdd(TKey, TValue)' method</value>
+  </data>
+  <data name="PreferDictionaryTryAddMessage" xml:space="preserve">
+    <value>Prefer a 'TryAdd' call over an 'Add' call guarded by a 'ContainsKey' check to avoid double lookup</value>
+  </data>
+  <data name="PreferDictionaryTryAddDescription" xml:space="preserve">
+    <value>Prefer a 'TryAdd' call over an 'Add' call guarded by a 'ContainsKey' check. 'TryAdd' behaves the same as 'Add' except that it doesn't throw an exception when a value for a given key already exists, but rather returns 'false'.</value>
```

Comment:
```suggestion
    <value>Prefer a 'TryAdd' call over an 'Add' call guarded by a 'ContainsKey' check. 'TryAdd' behaves the same as 'Add', except that when the specified key already exists, it returns 'false' instead of throwing an exception.</value>
```

Reply:
Done.